index.js

```diff
@@ -2613,6 +2613,42 @@
   }
   S.ifElse = def('ifElse', {}, [Pred(a), Fn(a, b), Fn(a, b), a, b], ifElse);
 
+  //# when :: (a -> Boolean) -> (a -> b) -> a -> b
```

Comment:
> we should choose one which is "correct"

Just my 2¢:

The only type-correct signature of `ifElse` is the one currently documented:

```haskell
ifElse :: (a -> Bool) -> (a -> b) -> (a -> b) -> a -> b
--                       ^------^    ^------^
--                           \          /
--                     have to be the same type
```

So when we fix one of the "branches" with `id :: a -> a`, we get:

```haskell
-- a == b
ifElse :: (a -> Bool) -> (a -> a) -> (a -> a) -> a -> a

-- with `id` as one of the "branches"
when :: (a -> Bool) -> (a -> a) -> a -> a
```

To represent a case where we might run "when" function or leave value as is, we'd need to have @safareli's type with `Either` and not rely on `ifElse` as implementation since they have different semantics.